File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    do
    {
        printf("Height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);

    for (int row = 0; row < height; row++)
    {
        //printing spaces
        for (int k = 0; k < height - row - 1; k++)
        {
            printf(" ");
        }

        //printing hashes
        for (int j = 0; j < row + 2; j++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: mario.py
from cs50 import get_int

# reprompts user to give an int value between 0 and 23
while True:
    print("height: ", end="")
    height = get_int()

    if height >= 0 and height < 24:
        break

for i in range(height):

    # does not print a newline after printing spaces, waits for hashes
    print(" " * (height - 1 - i), end="")
    print("#" * (2 + i))
